linera-execution/tests/test_execution.rs

```diff
@@ -429,6 +430,501 @@ async fn test_cross_application_error() -> anyhow::Result<()> {
+            ExecutionRuntimeConfig::Synchronous,
+        )
+        .await;
+
+    let mut applications = register_mock_applications(&mut view, 1).await?;
+    let (application_id, application) = applications
+        .next()
+        .expect("Missing caller mock application");
+
+    let destination_chain = ChainId::from(ChainDescription::Root(1));
+    let dummy_message = RawOutgoingMessage {
+        destination: Destination::from(destination_chain),
+        authenticated: false,
+        kind: MessageKind::Simple,
+        message: b"msg".to_vec(),
+    };
+
+    application.expect_call(ExpectedCall::execute_operation({
+        let dummy_message = dummy_message.clone();
+        move |_runtime, _context, _operation| {
+            Ok(RawExecutionOutcome::default().with_message(dummy_message))
+        }
+    }));
+
+    let context = OperationContext {
+        chain_id: ChainId::root(0),
+        height: BlockHeight(0),
+        index: 0,
+        authenticated_signer: None,
+        next_message_index: 0,
+    };
+    let mut tracker = ResourceTracker::default();
+    let policy = ResourceControlPolicy::default();
+    let outcomes = view
+        .execute_operation(
+            context,
+            Operation::User {
+                application_id,
+                bytes: vec![],
+            },
+            &policy,
+            &mut tracker,
+        )
+        .await?;
+
+    let application_description = view
+        .system
+        .registry
+        .describe_application(application_id)
+        .await?;
+    let registration_message = RawOutgoingMessage {
+        destination: Destination::from(destination_chain),
+        authenticated: false,
+        kind: MessageKind::Simple,
+        message: SystemMessage::RegisterApplications {
+            applications: vec![application_description],
+        },
+    };
+
+    assert_eq!(
+        outcomes,
+        &[
+            ExecutionOutcome::System(
+                RawExecutionOutcome::default().with_message(registration_message)
+            ),
+            ExecutionOutcome::User(
+                application_id,
+                RawExecutionOutcome::default().with_message(dummy_message)
+            )
+        ]
+    );
+
+    Ok(())
+}
+
+/// Tests if a message is scheduled to be sent while an application is handling a cross-application
+/// call.
+#[tokio::test]
+async fn test_message_from_cross_application_call() -> anyhow::Result<()> {
+    let mut state = SystemExecutionState::default();
+    state.description = Some(ChainDescription::Root(0));
+    let mut view =
+        ExecutionStateView::<MemoryContext<TestExecutionRuntimeContext>>::from_system_state(
+            state,
+            ExecutionRuntimeConfig::Synchronous,
+        )
+        .await;
+
+    let mut applications = register_mock_applications(&mut view, 2).await?;
+    let (caller_id, caller_application) = applications
+        .next()
+        .expect("Missing caller mock application");
```

Comment:
nit: Generally, we're trying to follow the pattern `.expect("something should be true")`

Reply:
Done.